packages/core/src/fhirpath/utils.ts

```diff
@@ -66,30 +66,35 @@ export function singleton(collection: TypedValue[], type?: string): TypedValue |
  * This function returns the value and the type.
  * @param input - The base context (FHIR resource or backbone element).
  * @param path - The property path.
+ * @param profileUrl - (optional) Profile URl
  * @returns The value of the property and the property type.
  */
-export function getTypedPropertyValue(input: TypedValue, path: string): TypedValue[] | TypedValue | undefined {
+export function getTypedPropertyValue(
+  input: TypedValue,
+  path: string,
+  profileUrl?: string
```

Comment:
Sensible, but I wonder if we should generalize this to `context` or `options` so that we don't end up with many extra args.

Separately, I'll follow up with you on an idea from @mattwiller  about a `Proxy` based solution.  I've been resistant to it, because it will add a layer of complexity, but changes like this make me wonder if it's necessary complexity.